### PR TITLE
Add TO Go delete deliveryservice_user

### LINKED ISSUE #3837: Rewrite /deliveryservice_user/{{DSID}}/{{userID}} to Go
## I'm submitting a ...
improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
- Documentation (possibly)
- Traffic Ops

## Current behavior:
`/deliveryservice_user/{{DSID}}/{{userID}}` is handled by Perl

## Expected / new behavior:
`/deliveryservice_user/{{DSID}}/{{userID}}` should be handled by Go.

### Minimum API Version
1.1

### Request Methods
- [ ] `DELETE`

## Minimal reproduction of the problem with instructions:
N/A

## Anything Else
PR in progress: #2456 . Progress to completion is unknown

### PR DESCRIPTION
**Pull request by rob05c**
Rewrites `/deliveryservice_user/{{DSID}}/{{userID}}` to Go. Closes #3837

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1893/
Test PASSed.

**Comment by rob05c** [comment]
API tests are in https://github.com/apache/trafficcontrol/pull/2339

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1948/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2000/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2054/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4149/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4150/
Test PASSed.

File: traffic_ops/traffic_ops_golang/tenant/tenancy.go [hunk 1]
```diff
@@ -58,6 +58,18 @@ func GetDeliveryServiceTenantInfo(xmlID string, tx *sql.Tx) (*DeliveryServiceTen
 	return &ds, nil
 }
 
+func GetDeliveryServiceTenantInfoID(tx *sql.Tx, dsID int) (*DeliveryServiceTenantInfo, error) {
```
Comment (mhoppa) [inline traffic_ops/traffic_ops_golang/tenant/tenancy.go]: I dont think this is used anywhere in this PR?

**Comment by ocket8888** [comment]
we shouldn't rewrite this using query instead of path parameters - that breaks the API contract.

**Comment by mhoppa** [comment]
This was decided to be deprecated as Delivery service users have no effect in our current system due to the introduction of tenancy. We can close this in place of opening a deprecation PR.

**Comment by mhoppa** [comment]
@ocket8888 @rob05c ^

**Comment by ocket8888** [comment]
Well if he wants he can still add deprecation notices to this. If that gets done before a deprecation PR is merged then we can merge this. If not, we can close this. Starting to look like that may be the case.

**Comment by mhoppa** [comment]
@ocket8888 lets close it now https://github.com/apache/trafficcontrol/pull/4367 :)

**Comment by ocket8888** [comment]
Yep. Resolved by #4367

**Pull request closed without merging**